### ISSUE TITLE
Apply `filtered` attribute to silenced users' statuses

### ISSUE THREAD

**Issue by trwnh** [mastodon/mastodon#20375]
### Pitch

Instead of just having a heuristic on whether the silenced user is followed or not, attach a `filtered` attribute with an account match.

### Motivation

Related: #18955, #11218, #9028

This would allow supporting different ideas of what it means to "silence" a user or instance, generally giving clients and users more control over their experience.

Silenced users currently are hidden entirely unless followed, but you may be willing to view posts from them anyway. Perhaps there is a domain-wide silencing and you find the risk from that domain's users to be acceptable. Or perhaps you wish for their replies to simply be deprioritized or hidden behind a warning rather than dropped entirely from your perspective.

Having a `filtered` attribute would allow clients to apply filter policies in a more consistent way, as well as supporting additional paradigms.

In summary, the following use cases would be made possible:

- "Quality filter" that shows silenced users at the bottom of the conversation, behind a "see more" fold
- Show in timelines, but hide behind a warning instead of dropping the status
- Any extensions to the filter policies could be applied in the future
